### ISSUE TITLE
Some precursors are being reported in the library output, even though they are not identified in the result output.

### ISSUE THREAD

**Issue by ec-ho-ra-mos** [vdemichev/DiaNN#1485]
I am using DIA-NN version 2.1.0. While generating a library, I noticed that the number of Precursor.Id entries in the library output is higher than in the result output. How is this possible? Is this behavior intentional? If DIA-NN can "suggest" a precursor in the library output but does not "identify" it in the result output, how reliable are those precursors?

**Comment by vdemichev** [comment]
Hi,

Can you please quote the numbers you get (and how you calculate them in each case) and share the log?

Best,
Vadim

**Comment by ec-ho-ra-mos** [comment]
Thank you very much for your prompt response.

The unique Precursor.Ids in my result output is **22835**, while it is **23059** in the library output. 

This is the log for this analysis:
DIA-NN 2.1.0 Academia  (Data-Independent Acquisition by Neural Networks)
Compiled on Mar 25 2025 10:59:54
Current date and time: Mon Apr  7 17:44:50 2025
CPU: GenuineIntel Intel(R) Xeon(R) CPU E3-1226 v3 @ 3.30GHz
SIMD instructions: AVX AVX2 FMA SSE4.1 SSE4.2
Logical CPU cores: 4
diann.exe --f [REDACTED_PATH_1] --f [REDACTED_PATH_2] --f [REDACTED_PATH_3] --f [REDACTED_PATH_4] --lib  --threads 4 --verbose 1 --out [REDACTED_OUTPUT_PATH_REPORT] --qvalue 0.01 --matrices --out-lib [REDACTED_OUTPUT_PATH_LIBRARY] --gen-spec-lib --predictor --fasta [REDACTED_FASTA_PATH] --fasta-search --min-fr-mz 400 --max-fr-mz 800 --met-excision --min-pep-len 7 --max-pep-len 30 --min-pr-mz 400 --max-pr-mz 800 --min-pr-charge 1 --max-pr-charge 4 --cut K*,R* --missed-cleavages 1 --unimod4 --rt-profiling

Thread number set to 4
Output will be filtered at 0.01 FDR
Precursor/protein x samples expression level matrices will be saved along with the main report
A spectral library will be generated
Deep learning will be used to generate a new in silico spectral library from peptides provided
DIA-NN will carry out FASTA digest for in silico lib generation
Min fragment m/z set to 400
Max fragment m/z set to 800
N-terminal methionine excision enabled
Min peptide length set to 7
Max peptide length set to 30
Min precursor m/z set to 400
Max precursor m/z set to 800
Min precursor charge set to 1
Max precursor charge set to 4
In silico digest will involve cuts at K*,R*
Maximum number of missed cleavages set to 1
Cysteine carbamidomethylation enabled as a fixed modification
The spectral library (if generated) will retain the original spectra but will include empirically-aligned RTs
WARNING: for DIA-NN to switch to the new .raw reader library, please download and install .NET Runtime 8.0.14 or later https://dotnet.microsoft.com/en-us/download/dotnet/8.0
DIA-NN will automatically optimise the mass accuracy for the first run of the experiment, use this mode for preliminary analyses only
WARNING: incorrect settings, the in silico-predicted library must be generated in a separate pipeline step and then used to process the raw data, now without activating FASTA digest

4 files will be processed
[0:00] Loading FASTA [REDACTED_FASTA_PATH]
[0:03] Processing FASTA
[0:05] Assembling elution groups
[0:09] 1976796 precursors generated
[0:09] Gene names missing for some isoforms
[0:09] Library contains 20377 proteins, and 20183 genes
[0:11] [0:17] [15:49] [18:42] [18:53] [18:53] Saving the library to [REDACTED_OUTPUT_PATH_LIBRARY].predicted.speclib
[18:56] Initialising library
[19:07] Loading spectral library [REDACTED_OUTPUT_PATH_LIBRARY].predicted.speclib
[19:09] Library annotated with sequence database(s): [REDACTED_FASTA_PATH]
[19:10] Spectral library loaded: 20377 protein isoforms, 29110 protein groups and 1976796 precursors in 1298612 elution groups.
[19:10] Loading protein annotations from FASTA [REDACTED_FASTA_PATH]
[19:10] Annotating library proteins with information from the FASTA database
[19:10] Gene names missing for some isoforms
[19:10] Library contains 20377 proteins, and 20183 genes
[19:12] Initialising library
WARNING: it is strongly recommended to enable MBR when analysing with a large library, if this is a quantitative analysis

[19:22] File #1/4
[19:22] Loading run [REDACTED_PATH_1]
[19:34] Pre-processing...
[19:38] 5336 MS1 and 133400 MS2 scans in 5336 (inferred) and 5336 (encoded) cycles, 639967 precursors in range
[19:39] Calibrating with mass accuracies 25 (MS1), 40 (MS2)
[20:41] Recommended MS1 mass accuracy setting: 22 ppm
[27:15] RT window set to 6.11041
[27:15] Peak width: 9.944
[27:15] Scan window radius set to 21
[27:15] Recommended MS1 mass accuracy setting: 18 ppm
[34:30] Optimised mass accuracy: 12 ppm
[36:51] Main search
[39:42] Removing low confidence identifications
[39:50] Removing interfering precursors
[39:53] Training neural networks on 13244 target and 7827 decoy PSMs
[40:40] Number of IDs at 0.01 FDR: 6163
[40:40] Calculating protein q-values
[40:41] Number of genes identified at 1% FDR: 2804 (precursor-level), 2609 (protein-level) (inference performed using proteotypic peptides only)
[40:41] Quantification
[40:42] Quantification information saved to [REDACTED_PATH_1].quant

[40:42] File #2/4
[40:42] Loading run [REDACTED_PATH_2]
[40:52] Pre-processing...
[40:55] 5345 MS1 and 133625 MS2 scans in 5345 (inferred) and 5345 (encoded) cycles, 532572 precursors in range
[40:57] Calibrating with mass accuracies 25 (MS1), 40 (MS2)
[41:26] Recommended MS1 mass accuracy setting: 27 ppm
[46:53] RT window set to 5.93349
[46:53] Recommended MS1 mass accuracy setting: 27 ppm
[49:19] Main search
[51:44] Removing low confidence identifications
[51:53] Removing interfering precursors
[51:56] Training neural networks on 15984 target and 9269 decoy PSMs
[52:45] Number of IDs at 0.01 FDR: 8381
[52:45] Calculating protein q-values
[52:46] Number of genes identified at 1% FDR: 3432 (precursor-level), 3174 (protein-level) (inference performed using proteotypic peptides only)
[52:46] Quantification
[52:46] Quantification information saved to [REDACTED_PATH_2].quant

[52:47] File #3/4
[52:47] Loading run [REDACTED_PATH_3]
[52:54] Pre-processing...
[52:57] 5341 MS1 and 133525 MS2 scans in 5341 (inferred) and 5341 (encoded) cycles, 427040 precursors in range
[52:59] Calibrating with mass accuracies 25 (MS1), 40 (MS2)
[53:41] Recommended MS1 mass accuracy setting: 25 ppm
[58:02] RT window set to 8.06916
[58:02] Recommended MS1 mass accuracy setting: 27 ppm
[60:26] Main search
[62:44] Removing low confidence identifications
[62:49] Removing interfering precursors
[62:52] Training neural networks on 9836 target and 5468 decoy PSMs
[63:24] Number of IDs at 0.01 FDR: 5341
[63:24] Calculating protein q-values
[63:25] Number of genes identified at 1% FDR: 2473 (precursor-level), 2329 (protein-level) (inference performed using proteotypic peptides only)
[63:25] Quantification
[63:25] Quantification information saved to [REDACTED_PATH_3].quant

[63:25] File #4/4
[63:25] Loading run [REDACTED_PATH_4]
[63:34] Pre-processing...
[63:37] 5314 MS1 and 132850 MS2 scans in 5314 (inferred) and 5314 (encoded) cycles, 377217 precursors in range
[63:39] Calibrating with mass accuracies 25 (MS1), 40 (MS2)
[64:45] Recommended MS1 mass accuracy setting: 29 ppm
[69:21] RT window set to 8.44319
[69:21] Recommended MS1 mass accuracy setting: 29 ppm
[71:44] Main search
[73:47] Removing low confidence identifications
[73:50] Removing interfering precursors
[73:53] Training neural networks on 6120 target and 3677 decoy PSMs
[74:20] Number of IDs at 0.01 FDR: 2950
[74:21] Calculating protein q-values
[74:21] Number of genes identified at 1% FDR: 1624 (precursor-level), 1547 (protein-level) (inference performed using proteotypic peptides only)
[74:21] Quantification
[74:22] Quantification information saved to [REDACTED_PATH_4].quant

[74:22] Cross-run analysis
[74:22] Reading quantification information: 4 files
[74:28] Quantifying peptides
WARNING: not enough peptides for normalisation
WARNING: QuantUMS requires 6 or more runs for the optimisation of its hyperparameters to perform best.
[74:28] Quantification parameters: 0.332807, 0.00703215, 0.00326131, 0.05, 0.05, 0.05, 0.05, 0.05, 0.05, 0.05, 0.05, 0.05, 0.05, 0.05, 0.05, 0.05
WARNING: not enough peptides for normalisation
[74:29] Assembling protein groups
[74:30] Quantifying proteins
[74:31] Calculating q-values for protein and gene groups
[74:31] Calculating global q-values for protein and gene groups
[74:31] Protein groups with global q-value <= 0.01: 4367
[74:32] Compressed report saved to [REDACTED_OUTPUT_PATH_REPORT].parquet. Use R 'arrow' or Python 'PyArrow' package to process
[74:32] Saving precursor levels matrix
[74:32] Precursor levels matrix (1% precursor and protein group FDR) saved to [REDACTED_OUTPUT_PATH_REPORT].pr_matrix.tsv.
[74:32] Saving protein group levels matrix
[74:32] Protein groups matrix saved to [REDACTED_OUTPUT_PATH_REPORT].pg_matrix.tsv.
[74:32] Saving gene group levels matrix
[74:32] Gene groups matrix saved to [REDACTED_OUTPUT_PATH_REPORT].gg_matrix.tsv.
[74:32] Saving unique genes levels matrix
[74:32] Unique genes matrix saved to [REDACTED_OUTPUT_PATH_REPORT].unique_genes_matrix.tsv.
[74:32] Manifest saved to [REDACTED_OUTPUT_PATH_REPORT].manifest.txt
[74:32] Stats report saved to [REDACTED_OUTPUT_PATH_REPORT].stats.tsv
[74:32] Generating spectral library:
[74:32] 22835 target and 227 decoy precursors saved
[74:32] Spectral library saved to [REDACTED_OUTPUT_PATH_LIBRARY].parquet

**Comment by vdemichev** [comment]
OK, those extras are decoys saved in the library, the value of the Decoy column is 1 for these.

Best,
Vadim

**Comment by ec-ho-ra-mos** [comment]
I just checked, and you're correct. The "extra" precursors in the library output appear to be decoys. I only realized this now, that DIA-NN is also outputting decoys in the library output. Has this been the case since version 1.8?

Thank you very much for your prompt response!

**Comment by vdemichev** [comment]
This was first implemented in 1.9-series, the purpose is more accurate FDR control with MBR.
